src/modules/flight_mode_manager/tasks/AutoFollowTarget/FlightTaskAutoFollowTarget.hpp

```diff
@@ -125,91 +161,189 @@ class FlightTaskAutoFollowTarget : public FlightTask
-		FOLLOW_PERSPECTIVE_BEHIND_LEFT_ANGLE_DEG = 225
+		FOLLOW_PERSPECTIVE_BEHIND_LEFT_ANGLE_DEG = -135
 	};
 
-	enum {
+	// Follow Altitude modes set by the parameter FLW_TGT_ALT_M
+	enum FollowAltitudeMode
+	{
 		FOLLOW_ALTITUDE_MODE_CONSTANT,
+		FOLLOW_ALTITUDE_MODE_TRACK_TERRAIN,
 		FOLLOW_ALTITUDE_MODE_TRACK_TARGET
 	};
 
-	enum {
-		FOLLOW_GIMBAL_MODE_2D,
-		FOLLOW_GIMBAL_MODE_2D_WITH_TERRAIN,
-		FOLLOW_GIMBAL_MODE_3D
-	};
+	/**
+	 * Update the Follow height based on RC commands
+	 *
+	 * If the drone's height error to the setpoint is within the an user adjustment error time window
+	 * follow_height will be adjusted with a speed proportional to user RC command
+	 *
+	 * @param sticks Sticks object to get RC commanded values for adjustments
+	 */
+	void update_rc_adjusted_follow_height(const Sticks &sticks);
 
 	/**
-	 * Get the current follow-me perspective setting from PX4 parameters
+	 * Update the Follow distance based on RC commands
 	 *
-	 * @param param_nav_ft_fs value of the parameter NAV_FT_FS
-	 * @return Angle [deg] from which the drone should view the target while following it, with zero degrees indicating the target's 12 o'clock
+	 * If the drone's distance error to the setpoint is within the an user adjustment error time window
+	 * follow_distance will be adjusted with a speed proportional to user RC command
+	 *
+	 * @param sticks Sticks object to get RC commanded values for adjustments
+	 * @param drone_to_target_vector Tracked follow distance variable reference which will be updated to the new value
+	 */
+	void update_rc_adjusted_follow_distance(const Sticks &sticks, const Vector2f &drone_to_target_vector);
+
+	/**
+	 * Update the Follow angle based on RC commands
+	 *
+	 * If the drone's orbit angle in relation to target is within the an user adjustment error time window
+	 * away from the orbit angle setpoint, follow_angle will be adjusted with a speed proportional to user RC command
+	 *
+	 * @param sticks Sticks object to get RC commanded values for adjustments
+	 * @param measured_angle Measured current drone's orbit angle around the target (depends on tracked target orientation for reference)
+	 * @param tracked_orbit_angle_setpoint Rate constrained orbit angle setpoint value from last command
+	 */
+	void update_rc_adjusted_follow_angle(const Sticks &sticks, const float measured_orbit_angle, const float tracked_orbit_angle_setpoint);
+
+	/**
+	 * Update the Second Order Target Position + Velocity Filter to track kinematically feasible target position and velocity
+	 *
+	 * @param follow_target_estimator Received value from alpha-beta-gamma target estimator filter output
+	 */
+	void update_target_position_velocity_filter(const follow_target_estimator_s &follow_target_estimator);
+
+	/**
+	 * Calculate the tracked target orientation and overwrite the tracked target orientation if necessary
+	 *
+	 * Note : Filtered target velocity is generated via 2nd order reference model filter can have overshooting behaviors
```

Comment:
Is there a link to any literature regarding "2nd order reference model"? I don't think this is a universal expression known to most developers.

Reply:
you could say something like "the target velocity is filtered through a second order multi-input multi-output system (reference model) and may overshoot when the natural frequency of the system is set to a low value" -- TMI? :)

Reply:
Still not sure about this, is MIMO more well known term for developers? Otherwise, I would just mention the library header location to say : "If you don't know about this reference model, check out this library"

Reply:
probably easiest is just to say "second order filter". It's just a second order dynamic system we use as a filter. I don't think you need to explain it in detail here. the reader just knows it is a filter, and can look up the implementation. aerospace controls people probably understand the term "reference model" more than others.

Reply:
Alrighty :) I opted for 'second order filter'